client/signup/steps/jpo-site-type/select-business-address.jsx

```diff
@@ -13,14 +14,79 @@ import Button from 'components/button';
 import FormLabel from 'components/forms/form-label';
 import FormTextInput from 'components/forms/form-text-input';
 
-class SelectBusinessAddress extends React.Component {
+const SelectBusinessAddress = React.createClass( {
 
-	static propTypes = {
+	propTypes: {
 		signupDependencies: PropTypes.object,
 		handleBusinessInfo: PropTypes.func,
 		businessInfo: PropTypes.object,
 		required: PropTypes.bool,
-	};
+	},
+
+	validateZipCode( zipCode ) {
+		return ! isNaN( parseFloat( zipCode ) ) && isFinite( zipCode );
```

Comment:
This is only allowing numerical zip codes, like those in USA. There are other countries where zip codes are alphanumerical.